src/SoftKiller.jl

```diff
@@ -0,0 +1,169 @@
+using JetReconstruction
+
+mutable struct SoftKiller <: TilingBase
+    _ymax::Float64
+    _ymin::Float64
+    _requested_drap::Float64
+    _requested_dphi::Float64
+    #selector 
+    _ntotal::Int64
+    _ngood::Int64
+    _dy::Float64
+    _dphi::Float64
+    _cell_area::Float64
+    _inverse_dy::Float64
+    _inverse_dphi::Float64
+    _ny::Int64
+    _nphi::Int64
+
+    function SoftKiller(rapmin::Float64, rapmax::Float64, drap::Float64, dphi::Float64)
+        print("4 variables \n")
+        grid = new(rapmax, rapmin, drap, dphi, 0, 0, 0.0, 0.0, 0.0, 0.0, 0.0, 0, 0)
+        _setup_grid(grid)
+        print(description(grid))
+        grid
+    end
+
+    function SoftKiller(rapmax::Float64, grid_size::Float64)
+        print("2 variables\n")
+        grid = new(rapmax, -rapmax, grid_size, grid_size, 0, 0, 0.0, 0.0, 0.0, 0.0, 0.0, 0,
+                   0)
+        _setup_grid(grid)
+        print(description(grid))
+        grid
+    end
+end
+
+tile_index(sk::SoftKiller,
+           p::PseudoJet)::Int64 = begin
+    y_minus_ymin = rapidity(p) - sk._ymin
+    if y_minus_ymin < 0
+        return -1
+    end
+
+    iy = round(Int64, y_minus_ymin * sk._inverse_dy)
+    if iy >= sk._ny
+        return -1
+    end
+
+    iphi = round(Int64, phi(p)*sk._inverse_dphi)
+    if iphi == sk._nphi
+        iphi = 0
+    end
+
+    res = round(Int64, iy*sk._nphi + iphi)
+
+    res + 1
+end
+
+function _setup_grid(sk::SoftKiller)
+    @assert sk._ymax > sk._ymin
+    @assert sk._requested_drap > 0
+    @assert sk._requested_dphi > 0
+
+    ny_double = (sk._ymax-sk._ymin) / sk._requested_drap
+    sk._ny = max(round(Int64, ny_double+0.5), 1)
+    sk._dy = (sk._ymax-sk._ymin) / sk._ny
+    sk._inverse_dy = sk._ny/(sk._ymax-sk._ymin)
+
+    sk._nphi = round(Int64, (2 * π) / sk._requested_dphi + 0.5)
+    sk._dphi = (2 * π) / sk._nphi
+    sk._inverse_dphi = sk._nphi/(2*π)
+
+    @assert sk._ny >= 1 and sk._nphi >= 1
+
+    sk._ntotal = sk._nphi * sk._ny
+    sk._cell_area = sk._dy * sk._dphi
+end
+
+description(sk::SoftKiller)::String = begin
+    #from definiton of is_initialised  in RectangularGrid.hh
+    if sk._ntotal <= 0
+        return "Uninitialised rectangular grid"
+    end
+
+    descr = "rectangular grid with rapidity extent $(sk._ymin) < rap < $(sk._ymax) \n total tiles  $(sk._ntotal) \n "
+    descr *= "tile size drap x dphi = $(sk._dy) x $(sk._dphi)"
+
+    #Selector implementation desn't exist 
+    descr
+end
+
+n_tiles(sk::SoftKiller)::Int64 = begin
```

Comment:
For `SoftKiller` is this useful? Usually you only need to write these accessor type functions if you are offering exposed functionality outside the package, or you need to do something special based on sub-types. It doesn't look like either of these is the case and in your code it's fine just to use `sk._ntotal` directly.

Please review other accessors using the same criteria!